starport_template/lib/stores/wallets_store.dart

```diff
@@ -97,7 +97,10 @@ class WalletsStore {
 
   final Observable<int?> _selectedWalletIndex = Observable(null);
 
-  set selectedWalletIndex(int? value) => Action(() => _selectedWalletIndex.value = value)();
+  set selectedWalletIndex(int? value) {
+    Action(() => _selectedWalletIndex.value = value)();
+    getBalances(selectedWallet.publicAddress);
```

Comment:
let's also check if the value has changed indeed and do the fetch only if it's different